pkg/reconciler/reconciler.go

```diff
@@ -1161,6 +1161,7 @@ func (r *Reconciler) setupWatches(mgr ctrl.Manager, c controller.Controller) err
 	if err := c.Watch(
 		source.Kind(mgr.GetCache(), secret),
 		handler.EnqueueRequestForOwner(mgr.GetScheme(), mgr.GetRESTMapper(), obj, handler.OnlyControllerOwner()),
+		preds...,
```

Comment:
Adding the preds to the secret watch and secrets do not have the label selector the secret watch is disabled for Centrals using the label selector.
In this case, all the Central belonging secrets must be labelled correctly.
Another approach would be to not rely on controller-runtime and add the skip to the reconcile function.

Reply:
`preds` added here not for the secrets, but for the `EnqueueRequestForOwner` event handler. This event handler uses `obj` which is built on top of Central GVK. So no need to label secrets

Reply:
Thanks for clarifying the fix, this makes sense.

Reply:
Changed fix implementation based on feedback